### ISSUE TITLE
No Authorization header in 1.4

### ISSUE THREAD

**Issue by billmcknight1953** [mgonto/restangular#887]
Restangular 1.4 has not solved the problem of sending an Authorization on every request.  That is, the following code:

RestangularProvider.setBaseUrl("https://url...");
RestangularProvider.setDefaultHeaders({'Authorization': 'Basic blabla=='});

simply fails to include Authorization header on the first request.

**Comment by dhaowoods** [comment]
Hi, @mgonto I have set the basic authenticaion using RestangularProvider.setDefaultHeaders({'Authorization':'Basic '+btoa("username:password")});
However, the first time I open my app that used ui.router and restangular, the browser prompt for username and password, but subsequent actions that cause API request to mysql's http plugin do not cause browser prompt, even when I reload or open it in new tab, or open a new browser window while the first browser window is still opened, it is just happened when I open totally new browser window process, and launch my app the first time,, how could my app be wrong?

**Comment by billmcknight1953** [comment]
So I found that the issue was that the Authorization header is null for the preflight requests, not that it was null for normal GET, PUT, POST etc requests.  Preflights are typical of CROSS DOMAIN situations.  The problem was that the cross-domain API was responding to a preflight request with a 401 challenge due to the lack of an Authorization header.  Since I had control of the API I made sure that the 401 challenge was not issued by the preflight request but only by the other (GET, PUT, POST) requests
